rust/src/helpers.rs

```diff
@@ -0,0 +1,4 @@
+extern crate failure;
+
+pub type Result<T> = ::std::result::Result<T, failure::Error>;
+pub type Hash = ::std::vec::Vec<u8>;
```

Comment:
while the leading colons on `::std` are not incorrect, they are not necessary either, and most of the time people omit them

Reply:
cool, the link as to the why (rust 2018) was good